inc/Config/DataSource/HttpDataSource.php

```diff
@@ -36,10 +36,17 @@ public function get_image_url(): ?string {
 		return null;
 	}
 
+	/**
+	 * Get the service name.
+	 */
+	public function get_service(): ?string {
+		return isset( $this->config['service'] ) ? $this->config['service'] : null;
```

Comment:
fyi PHP has nullish coalescing since 7.0
```suggestion
		return $this->config['service'] ?? null;
```